orchagent/dash/dashtunnelorch.cpp

```diff
@@ -0,0 +1,584 @@
+    auto it_nhop = nhop_oids.begin();
+    for (auto ip : ctxt.metadata.endpoints())
+    {
+        sai_object_id_t nhop_oid = *it_nhop++;
+        if (nhop_oid == SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_ERROR("Failed to create DASH tunnel next hop entry for tunnel %s, endpoint %s", tunnel_name.c_str(), to_string(ip).c_str());
+            continue;
+        }
+
+        if (parent_tunnel_removed)
+        {
+            SWSS_LOG_INFO("Removing tunnel next hop OID %" PRIx64" for failed DASH tunnel %s endpoint %s", nhop_oid, tunnel_name.c_str(), to_string(ip).c_str());
+            sai_status_t status = sai_dash_tunnel_api->remove_dash_tunnel_next_hop(nhop_oid);
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("Failed to remove DASH tunnel next hop OID %" PRIx64" for failed DASH tunnel %s endpoint %s", nhop_oid, tunnel_name.c_str(), to_string(ip).c_str());
+            }
+            continue;
+        }
+
+        DashTunnelEndpointEntry endpoint = { nhop_oid, SAI_NULL_OBJECT_ID };
+        tunnel_table_[tunnel_name].endpoints[to_string(ip)] = endpoint;
+        SWSS_LOG_INFO("Tunnel next hop entry added for tunnel %s, endpoint %s", tunnel_name.c_str(), to_string(ip).c_str());
+        addTunnelMember(tunnel_table_[tunnel_name].tunnel_oid, nhop_oid, ctxt);
+        remove_from_consumer = false; // if we add at least one tunnel member, tunnel needs to stay in consumer for tunnel member post-bulk ops
+    }
+    return remove_from_consumer;
+}
+
+void DashTunnelOrch::addTunnelMember(const sai_object_id_t tunnel_oid, const sai_object_id_t nhop_oid, DashTunnelBulkContext& ctxt)
+{
+    SWSS_LOG_ENTER();
+    std::vector<sai_attribute_t> tunnel_member_attrs;
+    sai_attribute_t tunnel_member_attr;
+
+    tunnel_member_attr.id = SAI_DASH_TUNNEL_MEMBER_ATTR_DASH_TUNNEL_ID;
+    tunnel_member_attr.value.oid = tunnel_oid;
+    tunnel_member_attrs.push_back(tunnel_member_attr);
+
+    tunnel_member_attr.id = SAI_DASH_TUNNEL_MEMBER_ATTR_DASH_TUNNEL_NEXT_HOP_ID;
+    tunnel_member_attr.value.oid = nhop_oid;
+    tunnel_member_attrs.push_back(tunnel_member_attr);
+
+    auto& member_object_ids = ctxt.tunnel_member_object_ids;
+    member_object_ids.emplace_back();
+    tunnel_member_bulker_.create_entry(&member_object_ids.back(), (uint32_t) tunnel_member_attrs.size(), tunnel_member_attrs.data());
+}
+
+bool DashTunnelOrch::addTunnelMemberPost(const std::string& tunnel_name, const DashTunnelBulkContext& ctxt)
+{
+    SWSS_LOG_ENTER();
+    bool remove_from_consumer = true;
+    const auto& member_oids = ctxt.tunnel_member_object_ids;
+    auto it_member = member_oids.begin();
+    for (auto ip : ctxt.metadata.endpoints())
+    {
+        sai_object_id_t member_oid = *it_member++;
+        if (member_oid == SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_ERROR("Failed to create DASH tunnel member entry for tunnel %s, endpoint %s", tunnel_name.c_str(), to_string(ip).c_str());
```

Comment:
Change to warning log, specify that we are continuing to do post-bulk for other tunnel members.